From c93836728e217f2d2d90c9ec32c3a5277c52d7f0 Mon Sep 17 00:00:00 2001
From: Mariaum <me@mariaum.com>
Date: Sat, 10 Aug 2019 13:17:29 -0300
Subject: [PATCH] Correctly parse URL ports at fromLegacyText.

It currently allows invalid ports, since validating numbers in Regex seems to be a nightmare.

diff --git a/chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java b/chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java
index b1c6df14..9672b7c1 100644
--- a/chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java
+++ b/chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java
@@ -17,7 +17,7 @@ import net.md_5.bungee.api.ChatColor;
 public final class TextComponent extends BaseComponent
 {
 
-    private static final Pattern url = Pattern.compile( "^(?:(https?)://)?([-\\w_\\.]{2,}\\.[a-z]{2,4})(/\\S*)?$" );
+    private static final Pattern url = Pattern.compile( "^(?:(https?)://)?([-\\w_\\.]{2,}\\.[a-z]{2,4}(?::[0-9]{1,5}+)?)(/\\S*)?$" );
 
     /**
      * Converts the old formatting system that used
-- 
2.17.1

